### ISSUE TITLE
Should notify for first ID of previously "unknown" observations

### ISSUE THREAD

**Issue by carrieseltzer** [inaturalist/INaturalistIOS#443]
If you upload an observation with no ID, there is no red indicator in the app when someone else adds the first ID. 

In my haste to try and upload observations before going out of cell range, I uploaded some without any IDs. Later I went back and added some coarse IDs in the app. I kept up on all of the notifications on my observations while I was gone (without my laptop), but didn't check notifications on the web for 9 days. I was surprised to find dashboard notifications for a handful of recent observations, such as [this one](https://www.inaturalist.org/observations/13595355#activity_comment_1867141). When I look at my observations in the app, I see no indication that someone added an ID (aside from the fact that there is a name there). 

![img_7928](https://user-images.githubusercontent.com/7501183/41851674-a95ac078-7856-11e8-9f9d-12f38a8f64d5.PNG)
![img_7927](https://user-images.githubusercontent.com/7501183/41851677-ac3018c0-7856-11e8-9969-d0329390a45c.PNG)

In short, observations in which another user adds the first ID show no count badge in the app, which makes it easy to miss activity on your observations if you're only watching the app.
